File: src/test/java/com/ferrpoint/agent/TestcontainersConfiguration.java
package com.ferrpoint.agent;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.testcontainers.containers.GenericContainer;
import org.testcontainers.utility.DockerImageName;

@TestConfiguration(proxyBeanMethods = false)
public class TestcontainersConfiguration {

    @Bean
    public GenericContainer<?> chromeContainer() {
        return new GenericContainer<>(DockerImageName.parse("chromedp/headless-shell:latest"))
            .withExposedPorts(9222);
    }
}

File: src/test/kotlin/com/ferrpoint/hermes/TestcontainersConfiguration.kt
package com.ferrpoint.hermes

import org.springframework.boot.test.context.TestConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.testcontainers.containers.GenericContainer
import org.testcontainers.utility.DockerImageName

@TestConfiguration(proxyBeanMethods = false)
class TestcontainersConfiguration {

    @Bean
    @ServiceConnection(name = "openwebui/ollama")
    fun ollamaContainer(): GenericContainer<*> {
        return GenericContainer(DockerImageName.parse("openwebui/ollama:latest"))
            .withExposedPorts(11434)
    }
}
